devicetypes/smartthings/zigbee-dual-switch.src/zigbee-dual-switch.groovy

```diff
@@ -0,0 +1,140 @@
+       }
+    // simulator metadata
+    simulator {
+        // status messages
+        status "on": "on/off: 1"
+        status "off": "on/off: 0"
+
+        // reply messages
+        reply "zcl on-off on": "on/off: 1"
+        reply "zcl on-off off": "on/off: 0"
+    }
+
+    tiles(scale: 2) {
+        multiAttributeTile(name:"switch", type: "lighting", width: 6, height: 4, canChangeIcon: true){
+            tileAttribute ("device.switch", key: "PRIMARY_CONTROL") {
+                attributeState "on", label:'${name}', action:"switch.off", icon:"st.switches.light.on", backgroundColor:"#00A0DC", nextState:"turningOff"
+                attributeState "off", label:'${name}', action:"switch.on", icon:"st.switches.light.off", backgroundColor:"#ffffff", nextState:"turningOn"
+                attributeState "turningOn", label:'${name}', action:"switch.off", icon:"st.switches.light.on", backgroundColor:"#00A0DC", nextState:"turningOff"
+                attributeState "turningOff", label:'${name}', action:"switch.on", icon:"st.switches.light.off", backgroundColor:"#ffffff", nextState:"turningOn"
+            }
+        }
+        standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+            state "default", label:"", action:"refresh.refresh", icon:"st.secondary.refresh"
+        }
+        main "switch"
+        details(["switch", "refresh"])
+    }
+}
+
+def installed() {
+         createChildDevices()
+         configure()
+}
+
+// Parse incoming device messages to generate events
+def parse(String description) {
+    log.debug "description is $description"
+    Map map = zigbee.getEvent(description)
+    if (map) {
+        if (description?.startsWith('on/off')) {
+            log.debug "receive on/off message without endpoint id"
+        } else {
+            Map descMap = zigbee.parseDescriptionAsMap(description)
+            log.debug "$descMap"
+
+            if (descMap?.clusterId == "0006" && descMap.sourceEndpoint == "01") {
+                def result = map ? createEvent(map) : [:]
+                sendEvent(result)
+             } else if (descMap?.clusterId == "0006" && descMap.sourceEndpoint == "02") {
+                def result = map ? createEvent(map) : [:]
+                def childDevice = childDevices.find {
+			it.deviceNetworkId == "$device.deviceNetworkId-ep2"
+                }
+                childDevice.sendEvent(result)
+	    }
+        }
+    }
+}
+
+private void createChildDevices() {
+       def i = 2
+       addChildDevice("Child Switch", "${device.deviceNetworkId}-ep${i}", null,
+                         [completedSetup: true, label: "${device.displayName.split("1")[-1]} ${i}",
+                         isComponent: false, componentName: "ch$i", componentLabel: "Channel $i"])
+        
+}
+
+private channelNumber(String dni) {
+        dni.split("-ep")[-1] as Integer
```

Comment:
if you change the endpoint names as Zach suggests above, you will also need to change this method